great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -33,23 +31,44 @@ class BatchRequestError(Exception):
     pass
 
 
-# For our year splitter we default the range to the last 2 year.
-_CURRENT_YEAR = datetime.now(dateutil.tz.tzutc()).year
-_DEFAULT_YEAR_RANGE = list(range(_CURRENT_YEAR - 1, _CURRENT_YEAR + 1))
-_DEFAULT_MONTH_RANGE = list(range(1, 13))
-
-
 @pydantic_dc.dataclass(frozen=True)
 class ColumnSplitter:
-    method_name: str
     column_name: str
-    # param_defaults is a Dict where the keys are the parameters of the splitter and the values are the default
-    # values are the default values if a batch request using the splitter leaves the parameter unspecified.
-    param_defaults: Dict[str, List] = pydantic.Field(default_factory=dict)
+    method_name: str
+    param_names: List[str]
 
-    @property
-    def param_names(self) -> List[str]:
-        return list(self.param_defaults.keys())
+    def param_defaults(self, data_asset: DataAsset) -> Dict[str, List]:
+        raise NotImplementedError
+
+
+@pydantic_dc.dataclass(frozen=True)
+class SqlYearMonthSplitter(ColumnSplitter):
+    method_name: str = "split_on_year_and_month"
+    param_names: List[str] = pydantic.Field(default_factory=lambda: ["year", "month"])
+
+    def param_defaults(self, data_asset: DataAsset) -> Dict[str, List]:
+        # This column splitter is only relevant to SQL data assets so we do some assertions
+        # to validate this.
+        from great_expectations.execution_engine import SqlAlchemyExecutionEngine
+
+        assert isinstance(data_asset, TableAsset), "data_asset must be a TableAsset"
+        assert isinstance(
+            data_asset.datasource.execution_engine, SqlAlchemyExecutionEngine
+        )
+        with data_asset.datasource.execution_engine.engine.connect() as conn:
+            # We opt to use a raw string instead of sqlalchemy.txt because we don't
+            # need any of the features for this query
+            # https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.text
+            col = self.column_name
+            q = f"select min({col}), max({col}) from {data_asset.table_name}"
```

Comment:
We hardcode the query now instead of using more advanced SqlAlchemy constructs. We can generalize this if we need to to support other databases.

Reply:
Same with the return value? Will we get a structured type or will it remain a dictionary?